hoomd/md/constrain.py

```diff
@@ -233,138 +98,180 @@ def set_params(self,rel_tol=None):
 
-    """
-    def __init__(self):
+    Set constituent particle types and coordinates for a rigid body.
 
-        # initialize the base class
-        ConstraintForce.__init__(self)
-
-        self.composite = True
-
-        # create the c++ mirror class
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            self.cpp_force = _md.ForceComposite(hoomd.context.current.system_definition)
-        else:
-            self.cpp_force = _md.ForceCompositeGPU(hoomd.context.current.system_definition)
-
-        hoomd.context.current.system.addCompute(self.cpp_force, self.force_name)
+    Args:
+        type_name (str): The type of the central particle
+        types (list): List of types of constituent particles
+        positions (list): List of relative positions of constituent
+            particles
+        orientations (list): List of orientations of constituent particles
+            (**optional**)
+        charge (list): List of charges of constituent particles
+            (**optional**)
+        diameters (list): List of diameters of constituent particles
+            (**optional**)
 
-    def set_param(self,type_name, types, positions, orientations=None, charges=None, diameters=None):
-        R""" Set constituent particle types and coordinates for a rigid body.
+    .. caution::
+        The constituent particle type must be exist.
+        If it does not exist, it can be created on the fly using
+        ``system.particles.types.add('A_const')``.
 
-        Args:
-            type_name (str): The type of the central particle
-            types (list): List of types of constituent particles
-            positions (list): List of relative positions of constituent particles
-            orientations (list): List of orientations of constituent particles (**optional**)
-            charge (list): List of charges of constituent particles (**optional**)
-            diameters (list): List of diameters of constituent particles (**optional**)
-
-        .. caution::
-            The constituent particle type must be exist.
-            If it does not exist, it can be created on the fly using
-            ``system.particles.types.add('A_const')``.
+    Example::
 
-        Example::
+        rigid = constrain.rigid()
+        rigid.set_param(
+            'A',
+            types = ['A_const', 'A_const'],
+            positions = [(0,0,1),(0,0,-1)]
+            )
+        rigid.set_param(
+            'B',
+            types = ['B_const', 'B_const'],
+            positions = [(0,0,.5),(0,0,-.5)]
+            )
 
-            rigid = constrain.rigid()
-            rigid.set_param('A', types = ['A_const', 'A_const'], positions = [(0,0,1),(0,0,-1)])
-            rigid.set_param('B', types = ['B_const', 'B_const'], positions = [(0,0,.5),(0,0,-.5)])
+    """
 
-        """
+    _cpp_class_name = "ForceComposite"
+    def __init__(
+        self,
+        type_name,
```

Comment:
`__init__` should not take in `type_name`, `types`, etc... arguments. These parameters will be set later. The only thing that `__init__` should do now is update the `param_dict` and later will define the type parameter body.